Allow the fontweight for the title to be customized

dylancom opened the pull request:
## Description

Currently you can customize the fontFamily and fontSize but not the fontWeight (of the header title).
This PR let's you also change the fontWeight of the header title.

WoLewicki commented:
Thanks for your contribution! 
1. In the PR template, there are all the places where the change/addition of a property should be added. It looks like you've missed the READMEs of `native-stack` and `createNativeStackNavigator`.
2. Can you add this change to Android too? It should be rather simple. You can check here: https://github.com/software-mansion/react-native-screens/blob/master/android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java#L236 and extend the `if` to check for both `weight` and `FontFamily`, only `FontFamily` (current), only `weight`.

If you need any help, please comment here.

dylancom commented:
@WoLewicki I tried Android but it had no effect, see: https://github.com/facebook/react-native/issues/26193.
Seems that if you want for e.g. a bold font, you need to specify a bold font family.
I updated the READMEs, improved the iOS code and added an example.

WoLewicki commented on README.md:
```diff
@@ -190,6 +190,10 @@ Customize the font family to be used for the title.
 
 Customize the size of the font to be used for the title.
 
+#### `titleFontWeight`
```
If it is not available on Android, please add the `(iOS only)` in all READMEs and other files with this.

WoLewicki commented on TestsExample/App.js:
```diff
@@ -20,10 +20,11 @@ import Test649 from './src/Test649';
 export default function App() {
-  return <Test42 />;
-};
+  return <Test706 />;
```
Please don't change the default test file.

WoLewicki commented on TestsExample/ios/Podfile.lock:
```diff
@@ -514,4 +514,4 @@ SPEC CHECKSUMS:
 PODFILE CHECKSUM: 123196e881422a7a7f3a9c228d6113e8de498360
 
-COCOAPODS: 1.9.3
+COCOAPODS: 1.10.0
```
We shouldn't mix normal changes with such ones in the same PR. Please change it back to what it was.

WoLewicki commented on ios/RNSScreenStackHeaderConfig.h:
```diff
@@ -17,6 +17,7 @@ typedef NS_ENUM(NSInteger, RNSStatusBarStyle) {
 @property (nonatomic, retain) NSString *title;
 @property (nonatomic, retain) NSString *titleFontFamily;
 @property (nonatomic, retain) NSNumber *titleFontSize;
+@property (nonatomic, retain) NSString *titleFontWeight;
```
I think we could add the same but for `largeTitle`, the changes should look pretty similar to the normal title.

WoLewicki reviewed (changes requested):
Looks good! Please apply the changes requested and this should be ready to merge then!

dylancom commented:
@WoLewicki I applied the requested changes.

WoLewicki commented on ios/RNSScreenStackHeaderConfig.m:
```diff
@@ -145,16 +145,18 @@ + (void)setAnimatedConfig:(UIViewController *)vc withConfig:(RNSScreenStackHeade
-      if (config.titleFontFamily) {
-        attrs[NSFontAttributeName] = [RCTFont updateFont:nil withFamily:config.titleFontFamily size:size weight:nil style:nil variant:nil scaleMultiplier:1.0];
+      NSString *weight = config.titleFontWeight ?: nil;
+      if (config.titleFontFamily || config.titleFontWeight) {
```
```suggestion
      if (family || weight) {
```

WoLewicki commented on ios/RNSScreenStackHeaderConfig.m:
```diff
@@ -163,14 +165,16 @@ + (void)setAnimatedConfig:(UIViewController *)vc withConfig:(RNSScreenStackHeade
-        if (config.largeTitleFontFamily) {
-          largeAttrs[NSFontAttributeName] = [RCTFont updateFont:nil withFamily:config.largeTitleFontFamily size:largeSize weight:nil style:nil variant:nil scaleMultiplier:1.0];
+        NSNumber *largeWeight = config.largeTitleFontWeight ?: nil;
+        if (config.largeTitleFontFamily || config.largeTitleFontWeight) {
```
```suggestion
        if (largeFamily || largeWeight) {
```

WoLewicki commented on ios/RNSScreenStackHeaderConfig.m:
```diff
@@ -291,32 +295,36 @@ + (UINavigationBarAppearance*)buildAppearance:(UIViewController *)vc withConfig:
-    if (config.titleFontFamily) {
-      attrs[NSFontAttributeName] = [RCTFont updateFont:nil withFamily:config.titleFontFamily size:size weight:nil style:nil variant:nil scaleMultiplier:1.0];
+    NSString *weight = config.titleFontWeight ?: nil;
+    if (config.titleFontFamily || config.titleFontWeight) {
```
same as above

WoLewicki commented on ios/RNSScreenStackHeaderConfig.m:
```diff
@@ -291,32 +295,36 @@ + (UINavigationBarAppearance*)buildAppearance:(UIViewController *)vc withConfig:
-    if (config.largeTitleFontFamily) {
-      largeAttrs[NSFontAttributeName] = [RCTFont updateFont:nil withFamily:config.largeTitleFontFamily size:largeSize weight:nil style:nil variant:nil scaleMultiplier:1.0];
+    NSString *largeWeight = config.largeTitleFontWeight ?: nil;
+    if (config.largeTitleFontFamily || config.largeTitleFontWeight) {
```
same as above

WoLewicki commented on src/index.d.ts:
```diff
@@ -239,6 +244,10 @@ declare module 'react-native-screens' {
      * @description Customize the size of the font to be used for the title.
      */
     titleFontSize?: number;
+    /**
+     * @description Customize the weight of the font to be used for the title.
+     */
```
```suggestion
    /**
     * @host (iOS only)
     * @description Customize the weight of the font to be used for the title.
     */
```

WoLewicki commented on src/native-stack/types.tsx:
```diff
@@ -194,6 +195,7 @@ export type NativeStackNavigationOptions = {
   headerTitleStyle?: {
     fontFamily?: string;
     fontSize?: number;
+    fontWeight?: string;
```
Please add this prop to the description above with a note (iOS only)

WoLewicki commented on src/native-stack/types.tsx:
```diff
@@ -262,9 +264,7 @@ export type NativeStackNavigationOptions = {
-export type NativeStackNavigatorProps = DefaultNavigatorOptions<
-  NativeStackNavigationOptions
-> &
+export type NativeStackNavigatorProps = DefaultNavigatorOptions<NativeStackNavigationOptions> &
```
Please revert it

WoLewicki reviewed (changes requested):
Some minor tweaks.

dylancom commented:
@WoLewicki Tweaked it.

WoLewicki commented on src/native-stack/types.tsx:
```diff
@@ -189,11 +190,13 @@ export type NativeStackNavigationOptions = {
    * Style object for header title. Supported properties:
    * - fontFamily
    * - fontSize
+   * - fontWeight
```
```suggestion
   * - fontWeight (iOS only)
```

WoLewicki commented:
Ok it looks like it is ready to be merged :tada: Thanks for your contribution!
